packages/fiber/src/core/hooks.tsx

```diff
@@ -62,7 +62,7 @@ export function useUpdate(callback: UpdateCallback, stage: StageTypes = Stages.U
   // Memoize ref
   const ref = useMutableCallback(callback)
   // Throw an error if a stage does not exist in the lifecycle
-  if (!stages.includes(stage)) throw `The stage does not exist in the lifecycle.`
+  if (!stages.includes(stage)) throw `An invoked stage does not exist in the lifecycle.`
```

Comment:
Consider throwing a full `Error` object here, instead of just a string.

```ts
throw(new Error("..."))
```

Both will yield backtrace information in the console (at least in Chrome), but it will make the error a little more obvious (and, if there ever is any tooling that intends to catch these, it will make its job a little easier.)

![image](https://user-images.githubusercontent.com/1061/172133333-e0f158e3-4c96-4c07-8782-acae77fcfebc.png)

Reply:
Thanks for the tip. I'm new to working with errors but this looks like a good enhancement.